azureml/pipelines/azureml_pipelines.py

```diff
@@ -305,3 +307,97 @@ def knn_fewshot(guidance_prog: Input, input_ds: Input, example_ds: Input):
     sub_pipeline = knn_fewshot(guidance_program, input_dataset, example_dataset)
 
     return sub_pipeline
+
+
+def create_random_fewshot_cot_pipeline(
```

Comment:
I think I need to come up with a better pattern for doing these pipelines. There's too much code duplication here still. However, that can happen after the merge